### PR TITLE
MEP12 API examples

### PR DESCRIPTION
**Pull request by NelleV**
MEP12-ifying API examples.

Related to #7206

### CODE REVIEW

File: examples/api/bbox_intersect.py [hunk 1]
```diff
@@ -1,3 +1,13 @@
+"""
+===========================================
+Changing colors of lines intersecting a box
+===========================================
+
+The lines interescting the rectangle are colored in red, while the other one
```
Comment (QuLogic) [inline examples/api/bbox_intersect.py]: inerescting -> intersecting
other one -> others

File: examples/api/collections_demo.py [hunk 2]
```diff
@@ -1,5 +1,7 @@
-'''Demonstration of LineCollection, PolyCollection, and
-RegularPolyCollection with autoscaling.
+'''
+=========================================================
+Line and Poly and RegularPoly Collection with autoscaling
```
Comment (QuLogic) [inline examples/api/collections_demo.py]: Line, Poly, and ...

File: examples/api/colorbar_basics.py [hunk 3]
```diff
@@ -1,3 +1,12 @@
+"""
+========
+Colorbar
+========
+
+This example shows how to use colorbar by specify the mappable object (here
```
Comment (QuLogic) [inline examples/api/colorbar_basics.py]: specify -> specifying

File: examples/api/compound_path.py [hunk 4]
```diff
@@ -1,5 +1,9 @@
 """
-Make a compund path -- in this case two simple polygons, a rectangle
+=============
+Compound path
+=============
+
+Make a compound path -- in this case two simple polygons, a rectangle
 and a triangle.  Use CLOSEOPOLY and MOVETO for the different parts of
```
Comment (QuLogic) [inline examples/api/compound_path.py]: There's an existing typo here; should be `CLOSEPOLY`.

File: examples/api/custom_projection_example.py [hunk 5]
```diff
@@ -1,3 +1,13 @@
+"""
+=================
+Custom projection
+=================
+
+This example showcases the Hammer projection by alleviating many features of
```
Comment (QuLogic) [inline examples/api/custom_projection_example.py]: I don't think alleviating is the right verb here; not sure what it's trying to say (utilizing/using would fit without changing the sentence, but doesn't quite imply the same things.)

File: examples/api/date_index_formatter.py [hunk 6]
```diff
@@ -1,4 +1,8 @@
 """
+====================================
+Custom tick formatter for timeseries
```
Comment (QuLogic) [inline examples/api/date_index_formatter.py]: timeseries -> time series
or change the description to match.

File: examples/api/date_index_formatter.py [hunk 7]
```diff
@@ -1,4 +1,8 @@
...
+====================================
+
 When plotting time series, e.g., financial time series, one often wants
 to leave out days on which there is no data, eh weekends.  The example
```
Comment (QuLogic) [inline examples/api/date_index_formatter.py]: The existing "eh" here is probably not someone being Canadian... Probably should be "e.g.," or maybe "i.e.,".

File: examples/api/engineering_formatter.py [hunk 8]
```diff
@@ -1,4 +1,8 @@
 '''
+=========================================
+Labeling ticks using engineering notations
```
Comment (QuLogic) [inline examples/api/engineering_formatter.py]: notations -> notation

File: examples/api/filled_step.py [hunk 9]
```diff
@@ -1,3 +1,12 @@
+"""
+=========================
+Hatched filled histograms
```
Comment (QuLogic) [inline examples/api/filled_step.py]: Hatched filled -> Hatch-filled

File: examples/api/mathtext_asarray.py [hunk 10]
```diff
@@ -1,5 +1,9 @@
 """
-Load a mathtext image as numpy array
+===============================
+A mathtext image as numpy array
+===============================
+
+This example shows how to make images from latex strings.
```
Comment (QuLogic) [inline examples/api/mathtext_asarray.py]: latex -> LaTeX

File: examples/api/quad_bezier.py [hunk 11]
```diff
@@ -1,3 +1,12 @@
+"""
+============
+Bezier Curve
+============
+
+This example showcases the PathPatch object to create a bezier polycurve path
```
Comment (QuLogic) [inline examples/api/quad_bezier.py]: bezier -> Bezier (or even better to include the accent Bézier)

File: examples/api/radar_chart.py [hunk 12]
```diff
@@ -1,5 +1,9 @@
 """
-Example of creating a radar chart (a.k.a. a spider or star chart) [1]_.
+=====================================
+Radar chart (aka spider or star chart
```
Comment (QuLogic) [inline examples/api/radar_chart.py]: Closing parenthesis missing.

File: examples/api/sankey_demo_old.py [hunk 13]
```diff
@@ -1,3 +1,11 @@
+"""
+========================
+Yet another Sankey chart
```
Comment (QuLogic) [inline examples/api/sankey_demo_old.py]: chart -> diagram

File: examples/api/sankey_demo_basics.py [hunk 14]
```diff
@@ -1,4 +1,9 @@
-"""Demonstrate the Sankey class by producing three basic diagrams.
+"""
+================
+The Sankey class
```
Comment (QuLogic) [inline examples/api/sankey_demo_basics.py]: 50/50 on changing class to diagram here too.

File: examples/api/sankey_demo_rankine.py [hunk 15]
```diff
@@ -1,4 +1,9 @@
-"""Demonstrate the Sankey class with a practicle example of a Rankine power
+"""
+===================
+Rankine power Cycle
```
Comment (QuLogic) [inline examples/api/sankey_demo_rankine.py]: Cycle -> cycle

File: examples/api/sankey_demo_rankine.py [hunk 16]
```diff
@@ -1,4 +1,9 @@
...
+===================
+
+Demonstrate the Sankey class with a practicle example of a Rankine power
```
Comment (QuLogic) [inline examples/api/sankey_demo_rankine.py]: practicle -> practical

File: examples/api/scatter_piecharts.py [hunk 17]
```diff
@@ -1,5 +1,9 @@
 """
-This example makes custom 'pie charts' as the markers for a scatter plotqu
+====================================
+Scatter plot with pie charts markers
```
Comment (QuLogic) [inline examples/api/scatter_piecharts.py]: charts -> chart

**Comment by NelleV** [comment]
Comments are addressed.

File: examples/api/custom_projection_example.py (hunk 5 above)
Comment (QuLogic) [inline examples/api/custom_projection_example.py]: Missed this one.

**Comment by tacaswell** [comment]
appveyor failure is network related.

**Comment by tacaswell** [comment]
backported to v2.x as a1b680bc1ed4f0305298e56ae915d74d08514a66

**Comment by tacaswell** [comment]
Thanks @NelleV

**Pull request merged**